core/src/main/java/com/linecorp/armeria/internal/common/logging/LoggingDecorators.java

```diff
@@ -39,6 +40,26 @@
 
     private LoggingDecorators() {}
 
+    /**
+     * Logs request and response using the specified {@code requestLogger} and {@code responseLogger}.
+     */
+    public static void logWhenComplete(
+            Logger logger, RequestContext ctx,
+            Consumer<RequestOnlyLog> requestLogger, Consumer<RequestLog> responseLogger) {
+        ctx.log().whenRequestComplete().thenAccept(requestLogger).exceptionally(e -> {
+            try (SafeCloseable ignored = ctx.push()) {
+                logger.warn("Unexpected exception while logging request: ", e);
```

Comment:
Probably better to add `ctx` to the log message?
```suggestion
                logger.warn("Unexpected exception while logging request: {}", ctx, e);
```

Reply:
`"{} Unexpected...", ctx, e` for consistency?

Reply:
Fixed. 😉